cli/resources.rs

```diff
@@ -239,18 +249,18 @@ impl Read for Resource {
 impl AsyncRead for Resource {
   fn poll_read(&mut self, buf: &mut [u8]) -> Poll<usize, Error> {
     let mut table = RESOURCE_TABLE.lock().unwrap();
-    let maybe_repr = table.get_mut(&self.rid);
-    match maybe_repr {
-      None => panic!("bad rid"),
```

Comment:
I think the panic is okay here. See above comments about `lookup()`.

Reply:
This panic is exactly the problem we're trying to fix. `poll_read` is called after lookup, but since it's a future it can be called after the resource is removed (see integration test I provided)

Reply:
Oh ok - sorry - I hadn't understood that. The problem is that we're implementing AsyncRead and that only returns std::io::Errors... I wonder if we can somehow to this without using AsyncRead, and return DenoError.